src/to-be-partially-checked.js

```diff
@@ -0,0 +1,54 @@
+import {matcherHint, printReceived} from 'jest-matcher-utils'
+import {checkHtmlElement} from './utils'
+
+export function toBePartiallyChecked(element) {
+  checkHtmlElement(element, toBePartiallyChecked, this)
+
+  const isValidInput = () => {
+    return (
+      element.tagName.toLowerCase() === 'input' && element.type === 'checkbox'
+    )
+  }
+
+  const isValidAriaElement = () => {
+    return (
+      element.getAttribute('role') === 'checkbox' &&
+      element.getAttribute('aria-checked') === 'mixed'
```

Comment:
This check is already done in `isPartiallyChecked` probably shouldn't be here, right?

Reply:
Yup, I don't think this should be part of this function. This function should only check if the element has a valid role for which this check makes sense. Then the actual check with the boolean value you set the `pass` property to, that's what then determines if the element is partially checked or not.